src/GafferScene/BoundQuery.cpp

```diff
@@ -0,0 +1,405 @@
+{
+	return getChild< Gaffer::StringPlug >( g_firstPlugIndex + 3 );
+}
+
+Gaffer::Box3fPlug* BoundQuery::boundPlug()
+{
+	return const_cast< Gaffer::Box3fPlug* >(
+		static_cast< BoundQuery const* >( this )->boundPlug() );
+}
+
+Gaffer::Box3fPlug const* BoundQuery::boundPlug() const
+{
+	return getChild< Gaffer::Box3fPlug >( g_firstPlugIndex + 4 );
+}
+
+Gaffer::AtomicBox3fPlug* BoundQuery::internalBoundPlug()
+{
+	return const_cast< Gaffer::AtomicBox3fPlug* >(
+		static_cast< BoundQuery const* >( this )->internalBoundPlug() );
+}
+
+Gaffer::AtomicBox3fPlug const* BoundQuery::internalBoundPlug() const
+{
+	return getChild< Gaffer::AtomicBox3fPlug >( g_firstPlugIndex + 5 );
+}
+
+Gaffer::V3fPlug* BoundQuery::centerPlug()
+{
+	return const_cast< Gaffer::V3fPlug* >(
+		static_cast< BoundQuery const* >( this )->centerPlug() );
+}
+
+Gaffer::V3fPlug const* BoundQuery::centerPlug() const
+{
+	return getChild< Gaffer::V3fPlug >( g_firstPlugIndex + 6 );
+}
+
+Gaffer::V3fPlug* BoundQuery::sizePlug()
+{
+	return const_cast< Gaffer::V3fPlug* >(
+		static_cast< BoundQuery const* >( this )->sizePlug() );
+}
+
+Gaffer::V3fPlug const* BoundQuery::sizePlug() const
+{
+	return getChild< Gaffer::V3fPlug >( g_firstPlugIndex + 7 );
+}
+
+void BoundQuery::affects( Gaffer::Plug const* const input, AffectedPlugsContainer& outputs ) const
+{
+	ComputeNode::affects( input, outputs );
+
+	if( input == internalBoundPlug() )
+	{
+		outputs.push_back( boundPlug()->minPlug()->getChild( 0 ) );
+		outputs.push_back( boundPlug()->minPlug()->getChild( 1 ) );
+		outputs.push_back( boundPlug()->minPlug()->getChild( 2 ) );
+		outputs.push_back( boundPlug()->maxPlug()->getChild( 0 ) );
+		outputs.push_back( boundPlug()->maxPlug()->getChild( 1 ) );
+		outputs.push_back( boundPlug()->maxPlug()->getChild( 2 ) );
+		outputs.push_back( centerPlug()->getChild( 0 ) );
+		outputs.push_back( centerPlug()->getChild( 1 ) );
+		outputs.push_back( centerPlug()->getChild( 2 ) );
+		outputs.push_back( sizePlug()->getChild( 0 ) );
+		outputs.push_back( sizePlug()->getChild( 1 ) );
+		outputs.push_back( sizePlug()->getChild( 2 ) );
+	}
+	else if(
+		( input == spacePlug() ) ||
+		( input == locationPlug() ) ||
+		( input == relativeLocationPlug() ) ||
+		( input == scenePlug()->boundPlug() ) ||
+		( input == scenePlug()->existsPlug() ) ||
+		( input == scenePlug()->transformPlug() ) )
+	{
+		outputs.push_back( internalBoundPlug() );
+	}
+}
+
+void BoundQuery::hash( Gaffer::ValuePlug const* const output, Gaffer::Context const* const context, IECore::MurmurHash& h ) const
+{
+	ComputeNode::hash( output, context, h );
+
+	if( output == internalBoundPlug() )
+	{
+		std::string const loc = locationPlug()->getValue();
+		if( ! loc.empty() )
+		{
+			ScenePlug const* const splug = scenePlug();
+			assert( splug != 0 );
+
+			Gaffer::BoolPlug const* const eplug = splug->existsPlug();
+			assert( eplug != 0 );
+
+			// NOTE : scene exists plug returns true by default when there is no input scene
```

Comment:
As discussed this morning, the fact that exists plug returns true when there is no input scene should likely be classified as a bug, and then this source query will be unnecessary.  We probably need to wait for John to weight in on fixing existsPlug though.